tests/features/data-visibility-and-review.feature

```diff
@@ -1,122 +1,202 @@
+  # - review_status has two states: "provisional" or "approved"
+  # - visibility and review_status are REQUIRED when creating new data
+  # - new data must use visibility and review_status (no new use of legacy fields)
+  # - legacy combined values will be migrated into visibility and review_status using a documented mapping
+  # - visibility and review must cover previous release_status business logic
+
+  Background:
+    Given a functioning api
+    And all database models have a visibility field that supports internal and public
+    And all database models have a review_status field that supports provisional and approved
+    And visibility and review_status are required fields when creating new records
+    And new records must use visibility and review_status as the source of truth
+    And legacy combined visibility/review fields are deprecated
+
+  # ---------------------------------------------------------------------------
+  # PUBLIC ACCESS
+  # ---------------------------------------------------------------------------
+
+  @public_access @visibility
+  Scenario Outline: Public users can only access public data
+    Given I am a public API consumer
+    And there is <data_type> data stored with visibility public or internal
+    When I request <data_type> data through unauthenticated endpoints
+    Then I should only see records that are marked visibility public
+    And records marked internal should not be returned
+    And the response should include the review_status for each public record
+
+    Examples:
+      | data_type    |
+      | locations    |
+      | samples      |
+      | observations |
+
+  # might not be relevant yet in NMSampleLocations
+  @public_access @reports
+  Scenario: Public reports include review status disclaimers
+    Given there are public observations with review_status provisional and approved
+    When a public user generates a report
+    Then all returned observations should have visibility public
+    And provisional observations should include a disclaimer derived from review_status
+    And approved observations should indicate that they are fully reviewed
+
+  # might not be relevant yet in NMSampleLocations
+  @public_access @data_download
+  Scenario: Public data downloads exclude internal datasets
+    Given a public user requests a CSV export
+    And the dataset contains public and internal records with different review_status values
+    When the download is prepared
+    Then only the public records should be included
+    And the download should list the review_status for each record
+
+  # ---------------------------------------------------------------------------
+  # AUTHENTICATED STAFF ACCESS
+  # ---------------------------------------------------------------------------
+
+  @staff_access @visibility
+  Scenario Outline: Staff can access all data and its review status
+    Given I am an authenticated staff member
+    When I view <data_type> data through internal endpoints
```

Comment:
We do, in a sense, have internal endpoints with `/docs-auth`, but they are still "public" if someone knows of that URL. I think that this should indicate if the user is authenticated or not, since that is what will ultimately determine if they have access to internal or public records